internal/models/sessions.go

```diff
@@ -85,8 +86,22 @@ func NewSession() (*Session, error) {
 	return session, nil
 }
 
-func FindSessionByID(tx *storage.Connection, id uuid.UUID) (*Session, error) {
+func FindSessionByID(tx *storage.Connection, id uuid.UUID, forUpdate bool) (*Session, error) {
 	session := &Session{}
+
+	if forUpdate {
+		// pop does not provide us with a way to execute FOR UPDATE
+		// queries which lock the rows affected by the query from
+		// being accessed by any other transaction that also uses FOR
+		// UPDATE
+		if err := tx.RawQuery(fmt.Sprintf("SELECT id FROM %q WHERE id = ? LIMIT 1 FOR UPDATE;", session.TableName()), id).Exec(); err != nil {
```

Comment:
i don't think the `LIMIT` clause is necessary since the id is the primary key here - also i think we can just to `tx.RawQuery(...).First(session)` which would also return the result and we don't have to query again below

Reply:
This is why I left the comment. You can't add the `FOR UPDATE` clause with Pop. It's just not there at all. This is what makes the magic happen. https://www.postgresql.org/docs/current/sql-select.html#SQL-FOR-UPDATE-SHARE

Reply:
LIMIT 1 is just defensive, if we migrate in the future to a compound key and forget about this, we won't be locking a ton of rows.